pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -560,6 +562,31 @@ func (p *DNSProxy) CheckAllowed(endpointID uint64, destPort uint16, destID ident
 	return false, nil
 }
 
+func configureConnection(conn *net.Conn, secId identity.NumericIdentity) error {
+	var file *os.File
+	var err error
+
+	switch l4conn := (*conn).(type) {
+	case *net.UDPConn:
+		if file, err = l4conn.File(); err != nil {
```

Comment:
It looks like `file` [should be closed](https://pkg.go.dev/net#UDPConn.File):
> File returns a copy of the underlying os.File. **It is the caller's responsibility to close f when finished.** Closing c does not affect f, and closing f does not affect c.
>
> The returned os.File's file descriptor is different from the connection's. Attempting to change properties of the original using this duplicate may or may not have the desired effect.

Reply:
Great catch! Thanks!